### ISSUE TITLE
external score

### ISSUE THREAD

**Issue by nat890194** [happyharrycn/actionformer_release#141]
Hello, I have two questions regarding the code:
1. Is the external score calculated based on occurrences in the training set? If so, why does the sum of the values for a single video exceed 1 in the downloaded features?
2. Does the code only handle training and validation, but not testing? If so, how can we use the code to make predictions on the test set after running eval.py?
Thank you for your help!

**Comment by tzzcl** [comment]
For the first question, you can refer to #19. 

For the second question, you can use the[ `saveonly`](https://github.com/happyharrycn/actionformer_release/blob/main/eval.py#L81) flag in eval.py to generate predictions only without evaluation

**Comment by nat890194** [comment]
Thanks for the instructions! The `saveonly` flag in eval.py works for generating predictions without evaluation.

Regarding the external score, I'm trying to understand it more clearly. Does the UntrimmedNet classifier only handle action classification (no time regression), and does it be an occurrence based on prediction class (i.e., does it simply predict which action occurs during the video)? Or is it more about an occurrence of predicting the "correct" of the action?

Additionally, I’m wondering when it’s better to use the external score. Should `multiclass_nms` always be set to False when we use the external score? What exactly is the purpose or suggestion behind using `multiclass_nms` or the external score?

Thanks again for your help!

**Comment by happyharrycn** [comment]
These external scores are generated by a classification model that predicts the probability of actions occurring in a given input video. In most practical scenarios, these scores are not necessary for consideration.

For your other question, multiclass_nms must be set to false when using external scores.

**Comment by nat890194** [comment]
Thanks for the explanation.
